### PR TITLE
add new ptau, dkg contract, interfaces folder

### PR DESCRIPTION
**Pull request by saiakilesh**

### CODE REVIEW

File: contracts/nouns/dkg.sol [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+        uint constant babyjub_sub_order = 2736030358979909402780800718157159386076813972158567259200215660948447373041;
+
+    IVerifierRound2 round2_verifier;
+
+    //
+    // Committee configuration
+    //
+
+    uint public n_comm;
+    uint public tally_threshold;
```
Comment (dtebbs) [inline contracts/nouns/dkg.sol:18]: Not part of your changes, but while we are touching this code, could we rename this to something generic like `threshold`?

File: contracts/nouns/nouns.sol [hunk 2]
```diff
@@ -85,53 +67,27 @@ contract Nouns {
...
-    event Round1Complete();
-    event Round2Share(uint indexed recip_id, uint sender_id, uint enc_sk_share, uint[2] enc_eph_PK);
     event TallyComplete(/*uint indexed vote_id, */ uint yay, uint nay, uint abstain);
 
     // Generator Point
     uint public constant Gx = 5299619240641551281634865583518297030282874472190772894086521144482721001553;
     uint public constant Gy = 16950150798460657717958625567821834550301663161624707787222815936182638968203;
 
     constructor(
+        address dkg_address, // DKG contract
```
Comment (dtebbs) [inline contracts/nouns/nouns.sol:82]: let's stay consistent with the leading `_` convention for the argument here.

Not sure if this is the best convention, and it's not used consistently everywhere (at some point we should fix a convention and choose a linter #19) but let's keep things locally consistent for now.

File: contracts/interfaces/IDkg.sol [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+interface IDkg {
+    function get_PK() external view returns (uint256, uint256);
+
+    function get_PK_for(uint participant_id) external view returns(uint, uint);
+
+    function get_committee_id_from_address(address addr) external view returns (uint);
+
+    function n_comm() external view returns (uint);
+}
```
Comment (dtebbs) [inline contracts/interfaces/IDkg.sol:12]: git shows a warning about no final newline, here and in a few other files.
(Worth setting your editor to always add this.)

**Review by dtebbs — CHANGES_REQUESTED** [review]
Just a preliminary review.  Overall looks great.
A few silly nits from me, some of which are just incidental - not caused directly by your changes.
Thanks.

File: contracts/nouns/nouns.sol (hunk 2 above)
Comment (saiakilesh) [reply contracts/nouns/nouns.sol]: Fixed

File: contracts/nouns/dkg.sol (hunk 1 above)
Comment (saiakilesh) [reply contracts/nouns/dkg.sol]: Fixed

File: contracts/interfaces/IDkg.sol (hunk 3 above)
Comment (saiakilesh) [reply contracts/interfaces/IDkg.sol]: Fixed

**Comment by dtebbs** [comment]
Looks good.  If you can let me know when you are done (or mark for re-review) I think we can merge.

**Comment by saiakilesh** [comment]
> Looks good. If you can let me know when you are done (or mark for re-review) I think we can merge.

I am done @dtebbs . Can merge whenever.

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -19,6 +19,13 @@ jobs:
           lfs: true
       - run: git lfs pull
 
+      - name: download ptau 16
+        run: |
+          cd circuits
```
[the remark this answers is not in the document]
Comment (saiakilesh) [reply .github/workflows/main.yml:22]: Sounds good. Will make the edit.

**Comment by dtebbs** [comment]
@saiakilesh this all looks good and seems like the demo works.
If you don't have any more changes, let me know or mark for rereview and I think we can merge.
I'll rebase my PR #23 on top of this to avoid conflicts.

**Comment by dtebbs** [comment]
Addresses #9

**Comment by saiakilesh** [comment]
> @saiakilesh this all looks good and seems like the demo works. If you don't have any more changes, let me know or mark for rereview and I think we can merge. I'll rebase my PR #23 on top of this to avoid conflicts.

@dtebbs I don't have any more changes. I am good to go for merge.

**Pull request merged**